geonode/qgis_server/views.py

```diff
@@ -740,3 +740,55 @@ def set_thumbnail(request, layername):
     }
     return HttpResponse(
         json.dumps(retval), content_type="application/json")
+
+
+def download_qgs(request, layername):
```

Comment:
I saw the method is very similaar with download_zip. The only different is the .qgs file included or excluded. Perhaps you can just extend the method of download_zip and add a parameter to include qgs file.

Reply:
yes indeed. i can add additional parameter to the GET request and filter them in the `url.py`. thanks for the comment @ismailsunni